joomla/joomla-cms
# [5.1] Fix for legacy editor-xtd plugins with array of buttons

Pull Request for Issue https://github.com/joomla/joomla-cms/issues/43128 .

### Summary of Changes

Fix compatibility with legacy editor-xtd  plugins, which return multiple buttons.


### Testing Instructions
Edit https://github.com/joomla/joomla-cms/blob/ef33d44375080b2e49962927fd3169523c3b610d/plugins/editors-xtd/contact/src/Extension/Contact.php#L46

And replace whole method with:
```php
public function onDisplay($name)
{
  return [
    new CMSObject([
        'action'  => 'insert',
        'icon'    => 'foobar',
        'text'    => 'Beer',
        'options' => ['content' => '🍺 ']
    ]),
    new CMSObject([
        'action'  => 'insert',
        'icon'    => 'foobar',
        'text'    => 'Cat',
        'options' => ['content' => '🐈 ']
    ]),
  ];
}
```

Go to content editing. And check for this button.


### Actual result BEFORE applying this Pull Request
Only Cat button is shown.


### Expected result AFTER applying this Pull Request
Both buttons is shown.


### Link to documentations
Please select:
